modules/tensor_mechanics/doc/content/modules/tensor_mechanics/tutorials/introduction/step04.md

```diff
@@ -0,0 +1,143 @@
+# Step 4 - Multiple submeshes
+
+As a final step in the tensor mechanics section of this tutorial we'll set up a
+pair of mesh separate mesh blocks - two cantilevers side by side, fixed at the
+bottom, that will be pushed towards eachother with pressure boundary conditions.
+
+!listing modules/tensor_mechanics/tutorials/introduction/step04.i
+
+## Input file
+
+### `Mesh`
+
+We again are using multiple mesh generators to set up a more complex mesh. You
+could also use a meshing software, such as Cubit or Gmsh to create your geometry
+and mesh - and will most definitely have to do so when modeling complex
+geometries for which no specialized mesh generators are provided. It is crucial
+to assign block and boundary IDs and names to your meshes do you can apply
+material models and boundary conditions.
+
+Here we are using the familiar
+[`GeneratedMeshGenerator`](GeneratedMeshGenerator.md), but with a few more
+options than last time. Note how we're specifying the extent of the mesh with
+the [!param](/Mesh/GeneratedMeshGenerator/xmin) and
+[!param](/Mesh/GeneratedMeshGenerator/xmax) parameters (we keep the default
+value for [!param](/Mesh/GeneratedMeshGenerator/ymin), which is 0). We also use
+the [!param](/Mesh/GeneratedMeshGenerator/bias_y) parameter to have the element
+height in the mesh gradually shrink towards the top (you will see later why that
+is a good idea). [!param](/Mesh/GeneratedMeshGenerator/boundary_name_prefix)
+adds a prefix string to the names of the default boundaries that the
+GeneratedMeshGenerator sets up (`left`, `right`, `top`, `bottom` (for >=2D),
+`front`, and `back` (for 3D)). The `generated` block will create a mesh with the
+boundaries `pillar1_left`, `pillar1_right`, `pillar1_top`, and `pillar1_bottom`.
+
+The `generated2` generator will set up the second pillar and works very much like the first, except that we are also adding the [!param](/Mesh/GeneratedMeshGenerator/boundary_id_offset) to ensure that the boundaries of the second pillar have their own unique IDs (this is done by shifing the default IDs of the four boundaries by 4 so they don't overlap with the IDs of pillar1).
+
+!alert note
+When modeling the deformation/bending of walls with solid elements make sure to
+mesh at least about *five* elements through the wall thickness!
+
+The [`MeshCollectionGenerator`](MeshCollectionGenerator.md) then combines the two separate pillar mesh objects into a single mesh object that contains all elements from both meshes. Note that this does **not** mean the two meshes are glued together or connected in any way! Also, in contrast to the previous step the mesh generators do not form a *chain*, but they form a tree structure. The important thing for MOOSE is that a single genrator sits at the end of such a chain or root of such a tree (otherwise MOOSE will complain with an error message).
```

Comment:
```suggestion
The [`MeshCollectionGenerator`](MeshCollectionGenerator.md) then combines the two separate pillar mesh objects into a single mesh object that contains all elements from both meshes. Note that this does **not** mean the two meshes are glued together or connected in any way! Also, in contrast to the previous step the mesh generators do not form a *chain*, but they form a tree structure. The important thing for MOOSE is that a single generator sits at the end of such a chain or root of such a tree (otherwise MOOSE will complain with an error message).
```